C/sort/bucketSort.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * C Program to Sort Array using Bucket Sort
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+/* Structing a Bucket*/
+struct bucket
+{
+    int count;
+    int *value;
+};
+
+/* Function for comparing Integers */
+int compareIntegers(const void *first, const void *second)
```

Comment:
I think Bucket sort can be implemented without using CompareIntegers function as well as with the lesser lines of code.

Try to make the code small and clean as well. There are a lot of comments which are not necessary.